teps/0090-matrix.md

```diff
@@ -595,12 +622,510 @@ For example:
+  * `Parameter` type must be `Array`.
+  * `Parameter` can be in only one of `matrix` or `params`, not both.
+  * `Parameter` must not have `Result` references.
+  * `Parameter` must substitute valid variables, such as `Execution Status`. 
+
+#### Phase 2: Execute TaskRuns
+* Implement fanning out of `PipelineTasks` into `TaskRuns` based on the `matrix`.
+
+#### Phase 3: Execute Runs
+* Implement fanning out of `PipelineTasks` into `Runs` based on the `matrix`.
+
+#### Phase 4: Consume Results
+* Support `Results` in the `Matrix`.
+* Support use of `Results` from `Tasks` fanned out by the `Matrix` in subsequent `Tasks`.
+
+## Design Evaluation
+
+### API Conventions
+
+In the proposed design, we comply with the [Kubernetes API conventions][k8s-api] such as:
+* Lists of named subobjects preferred over maps ([convention][k8s-api-objects]): we use 
+  named subobjects - `Parameters` - in the `matrix` instead of maps.
+* Think twice about `bool` fields ([convention][k8s-api-primitives]): we didn't use `bools`
+  to plan for future expansions - see [alternative](#1-boolean-in-parameter-specification) 
+  using `bools`.
+
+### Reusability
+
+* Existing features are reused instead of adding new ones, such as `Parameters`.
+* At `Pipeline` authoring time, authors can specify the `matrix` used to fan out the
+  `PipelineTask`. At `Pipeline` run time, users can control the execution as needed 
+  without modifying the `Pipeline` because the `matrix` allows variable substitution.
+
+### Simplicity
+
+* Provided the bare minimum features needed to solve the [use cases](#use-cases).
+  For example, we won't fanning out based on `Parameters` of type `Array` as 
+  discussed [above](#what-about-substituting-array-parameters-in-tasks). 
+* The structure and behavior of `matrix` is consistent with the existing `params`
+  field, making learnability easy and promoting adoption of the feature.
+
+### Flexibility
+
+* The proposed design supports future expansions, including those discussed in the
+  [future work](#future-work) below. For example, we can support implicit mapping
+  of `Parameters` and consuming `Results` from dynamic matrixed `PipelineTasks`.
+* The proposed design is aligned with ongoing work on the same components, such 
+  as `Parameters` and `Results` in [TEP-0075][tep-0075] and [TEP-0076][tep-0076].
+
+### Conformance
+
+* The proposed change is backwards compatible. 
+* The `matrix` field is optional, per the guidance in the *Tekton Pipelines*
+  [API Spec][api-spec].
+
+## Future Work
+
+These will be explored in their own TEPs later:
+* Support consuming `Results` from dynamic matrixed `PipelineTasks`.
+  * This is dependent on [TEP-0076][tep-0076] which has not yet been accepted.
+* Support implicit mapping of `Parameters` in `matrix`.
+  * This is still an `alpha` feature, we will wait until it's `beta` to add support.
+* Gather and incorporate user feedback.
+* Promote to `beta` and `V1`.
+
+## Alternatives
```

Comment:
another alternative could be the [github actions syntax](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions), where the matrix is defined at the pipeline level. I think this is a pretty nice option.